sql/wsrep_encryption.h

```diff
@@ -0,0 +1,71 @@
+/* Copyright 2019 Codership Oy <info@codership.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef WSREP_ENCRYPTION_H
+#define WSREP_ENCRYPTION_H
+
+#include "wsrep/encryption_service.hpp"
+
+/**
+ * Set encryption key. Serialize it and send to provider.
+ * 
+ * @param key Pointer to encryption key
+ * @param size Length of encryption key
+ * @param version Key version used
+ */
+int wsrep_set_encryption_key(const void* key, size_t size,
+                             unsigned int version);
+
+/**
+ * Serialize key_version and key into buffer
+ * 
+ * @param input Container for serialized key
+ * @param key Pointer to encryption key
+ * @param size Size of key
+ * @param version Key version
+ */
+void wsrep_key_serialize(std::vector<unsigned char>& input, const void* key,
```

Comment:
`wsrep_key_serialize()` is used only in `wsrep_encryption.cc`. This declaration is therefore not necessary. Documentation comments should be moved into `wsrep_encryption.cc` and the function should be made static.

Reply:
Good catch, will change.